### ISSUE TITLE
Renderer function should be called with the element as the this

### ISSUE THREAD

**Issue by matthewp** [matthewp/haunted#26]
Currently the renderer receives the element as the first and only argument. It makes sense to also provide it as the `this` value. This way you can destruct for properties in the argument and use `this` for the element if needed for imperative usage. For example:

```js
function App({ open }) {
  useEffect(() => {
    this.querySelector('div').addEventListener ...
  }, []);

  return html`<div class="${open ? 'open' : ''}>...</div>
}
```

**Comment by askbeka** [comment]
This will assume that renderer is not an arrow function.

**Comment by matthewp** [comment]
Fixed by #28
